patches/chrome-browser-profiles-profile.cc.patch

```diff
@@ -11,3 +19,11 @@ index ca36c316828f3d5ab3fb511d73cb975a86286c5e..b435bcf5ad55a48a21aa545d82553deb
        user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
  #if defined(OS_ANDROID)
    registry->RegisterStringPref(
+@@ -371,6 +372,7 @@ bool Profile::IsRegularProfile() const {
+ }
+ 
+ bool Profile::IsIncognitoProfile() const {
++  BRAVE_IS_INCOGNITO_PROFILE
```

Comment:
Seems like this one can be done via a `chromium_src` override.

Reply:
fixed in https://github.com/brave/brave-core/pull/7069/commits/afe2f417bc6b2122d106dd620390356c671118f8